presto-main/src/main/java/com/facebook/presto/operator/scalar/Indeterminate.java

```diff
@@ -0,0 +1,284 @@
+
+public class Indeterminate
+        extends SqlOperator
+{
+    public static final Indeterminate INDETERMINATE_OPERATOR = new Indeterminate();
+
+    private static final AtomicLong NEXT_VARIABLE_ID = new AtomicLong();
+
+    private Indeterminate()
+    {
+        super(INDETERMINATE, ImmutableList.of(typeVariable("T")), ImmutableList.of(), BOOLEAN.getTypeSignature(), ImmutableList.of(parseTypeSignature("T")));
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(arity == 1, "Expected arity to be 1");
+        Type type = boundVariables.getTypeVariable("T");
+        Class<?> indeterminateOperatorClass = generateIndeterminate(type, functionRegistry);
+        MethodHandle methodHandle = methodHandle(indeterminateOperatorClass, "indeterminate", type.getJavaType());
+        return new ScalarFunctionImplementation(false, ImmutableList.of(false), methodHandle, isDeterministic());
+    }
+
+    private static Class<?> generateIndeterminate(Type type, FunctionRegistry functionRegistry)
+    {
+        CallSiteBinder binder = new CallSiteBinder();
+
+        // Embed the MD5 hash code of input and output types into the generated class name instead of the raw type names,
+        // which could prevent the class name from hitting the length limitation and invalid characters.
+        byte[] md5Suffix = Hashing.md5().hashBytes(type.toString().getBytes()).asBytes();
+
+        ClassDefinition definition = new ClassDefinition(
+                a(PUBLIC, FINAL),
+                CompilerUtils.makeClassName(Joiner.on("$").join("Indeterminate", BaseEncoding.base16().encode(md5Suffix))),
+                type(Object.class));
+
+        Parameter value = arg("value", type.getJavaType());
+
+        MethodDefinition method = definition.declareMethod(
+                a(PUBLIC, STATIC),
+                "indeterminate",
+                type(boolean.class),
+                value);
+
+        Scope scope = method.getScope();
+        BytecodeBlock body = method.getBody();
+
+        Variable wasNull = scope.declareVariable(boolean.class, "wasNull");
+        body.append(wasNull.set(constantFalse()));
+
+        CachedInstanceBinder cachedInstanceBinder = new CachedInstanceBinder(definition, binder);
+
+        LabelNode end = new LabelNode("end");
+        List<Type> fieldTypes = type.getTypeParameters();
+        boolean unknownElement = fieldTypes.stream().anyMatch(t -> t.equals(UNKNOWN));
+        if (isComplexType(type) && unknownElement) {
+            // if the current field type is UNKNOWN which means this field is null, directly return true
+            body.push(true)
+                    .append(wasNull.set(constantFalse()))
+                    .gotoLabel(end);
+        }
+        else {
+            switch (type.getTypeSignature().getBase()) {
+                case ARRAY: {
```

Comment:
This approach is not great. It's not extensible to other types. Instead, each type should have its own implementation of the operator, in the same way every other operator works (e.g., ArrayHashCodeOperator vs RowHashCodeOperator vs Bigint.hashCode)